<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TreeEditor</class>
 <widget class="QDialog" name="TreeEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Edit Tree</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QGraphicsView" name="tree_display">
     <property name="renderHints">
      <set>QPainter::Antialiasing|QPainter::HighQualityAntialiasing|QPainter::TextAntialiasing</set>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QGroupBox" name="edit_nodes_groupbox">
     <property name="title">
      <string>Edit Nodes</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="current_label">
        <property name="text">
         <string>Content</string>
        </property>
        <property name="buddy">
         <cstring>current_editor</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="NodeInput" name="current_editor">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>64</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QPushButton" name="add_left_button">
        <property name="text">
         <string>Add Left Child</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QPushButton" name="add_right_button">
        <property name="text">
         <string>Add Right Child</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QPushButton" name="delete_current_button">
        <property name="text">
         <string>Delete Node</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QDialogButtonBox" name="button_box">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
     <property name="centerButtons">
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>NodeInput</class>
   <extends>QLineEdit</extends>
   <header>gui.widgets.NodeInput</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>button_box</sender>
   <signal>accepted()</signal>
   <receiver>TreeEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_box</sender>
   <signal>rejected()</signal>
   <receiver>TreeEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>add_left_button</sender>
   <signal>clicked()</signal>
   <receiver>TreeEditor</receiver>
   <slot>_on_clicked_add_left()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>324</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>add_right_button</sender>
   <signal>clicked()</signal>
   <receiver>TreeEditor</receiver>
   <slot>_on_clicked_add_right()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>324</x>
     <y>116</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>delete_current_button</sender>
   <signal>clicked()</signal>
   <receiver>TreeEditor</receiver>
   <slot>_on_clicked_delete_current()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>324</x>
     <y>149</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>current_editor</sender>
   <signal>textEdited(QString)</signal>
   <receiver>TreeEditor</receiver>
   <slot>_on_edited_current(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>349</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>_on_clicked_add_left()</slot>
  <slot>_on_clicked_add_right()</slot>
  <slot>_on_clicked_delete_current()</slot>
  <slot>_on_edited_current(QString)</slot>
 </slots>
</ui>
